openhtf/core/test_state.py

```diff
@@ -251,16 +250,26 @@ def finalize_normally(self):
     if self._is_aborted():
       return
 
-    allowed_outcomes = {measurements.Outcome.PASS}
-    if conf.allow_unset_measurements:
-      allowed_outcomes.add(measurements.Outcome.UNSET)
+    all_skipped = bool(self.test_record.phases)
+    any_failed = False
 
-    if any(meas.outcome not in allowed_outcomes
-           for phase in self.test_record.phases
-           for meas in phase.measurements.itervalues()):
+    for phase in self.test_record.phases:
+      if phase.outcome == test_record.PhaseOutcome.FAIL:
+        any_failed = True
+        all_skipped = False
+      elif phase.outcome != test_record.PhaseOutcome.SKIP:
+        all_skipped = False
+
+    if any_failed:
       self._finalize(test_record.Outcome.FAIL)
+    elif all_skipped:
+      self.logger.error('All phases were skipped, outcome ERROR.')
+      self.test_record.add_outcome_details(
+          'all_skipped', 'All phases were unexpectedly skipped.')
```

Comment:
We typically make the codes `ALL_CAPS` to distinguish from `Exceptions` and `measurement_names`